services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-submission-status/notice-of-intent-submission-status.service.ts

```diff
@@ -0,0 +1,144 @@
+import { ServiceNotFoundException } from '@app/common/exceptions/base.exception';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import * as dayjs from 'dayjs';
+import * as timezone from 'dayjs/plugin/timezone';
+import * as utc from 'dayjs/plugin/utc';
+import { Repository } from 'typeorm';
+import { NoticeOfIntentSubmission } from '../../../portal/notice-of-intent-submission/notice-of-intent-submission.entity';
+import { NoticeOfIntentSubmissionStatusType } from './notice-of-intent-status-type.entity';
+import { NOI_SUBMISSION_STATUS } from './notice-of-intent-status.dto';
+import { NoticeOfIntentSubmissionToSubmissionStatus } from './notice-of-intent-status.entity';
+
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+@Injectable()
+export class NoticeOfIntentSubmissionStatusService {
+  constructor(
+    @InjectRepository(NoticeOfIntentSubmissionToSubmissionStatus)
+    private statusesRepository: Repository<NoticeOfIntentSubmissionToSubmissionStatus>,
+    @InjectRepository(NoticeOfIntentSubmissionStatusType)
+    private submissionStatusTypeRepository: Repository<NoticeOfIntentSubmissionStatusType>,
+    @InjectRepository(NoticeOfIntentSubmission)
+    private applicationSubmissionRepository: Repository<NoticeOfIntentSubmission>,
```

Comment:
Rename to `noticeOfIntentSubmissionRepository` or something without application

Reply:
Done!